sleap/io/dataset.py

```diff
@@ -726,8 +727,19 @@ def extract(self, inds) -> "Labels":
             suggestions=self.suggestions,
             provenance=self.provenance,
         )
+        if copy:
+            new_labels = new_labels.copy()
         return new_labels
 
+    def copy(self) -> "Labels":
+        """Return a full deep copy of the labels.
+        Notes:
+            All objects will be re-created by serializing and then deserializing the
+            labels. This may be slow and will create new instances of all data
+            structures.
+        """
+        return type(self).from_json(self.to_json())
```

Comment:
have you considered using deepcopy instead of serializing/deserializing that will be slow indeed? https://docs.python.org/3/library/copy.html#copy.deepcopy